src/com/goide/inspections/GoStringIndexIsByteInspection.java

```diff
@@ -0,0 +1,105 @@
+
+public class GoStringIndexIsByteInspection extends GoInspectionBase {
+
+  private static final String TEXT_HINT = "Mismatched types: byte and string";
+  private static final GoStringIndexIsByteQuickFix STRING_INDEX_IS_BYTE_QUICK_FIX = new GoStringIndexIsByteQuickFix();
+
+  @NotNull
+  @Override
+  protected GoVisitor buildGoVisitor(@NotNull ProblemsHolder holder, @NotNull LocalInspectionToolSession session) {
+    return new GoVisitor() {
+
+      @Override
+      public void visitConditionalExpr(@NotNull GoConditionalExpr o) {
+        if (o.getEq() == null) {
+          return;
+        }
+
+        if (o.getLeft() instanceof GoIndexOrSliceExpr && o.getRight() instanceof GoStringLiteral) {
+          GoIndexOrSliceExpr left = (GoIndexOrSliceExpr)o.getLeft();
+          GoStringLiteral right = (GoStringLiteral)o.getRight();
+          checkTypeMismatch(o, holder, left, right, STRING_INDEX_IS_BYTE_QUICK_FIX);
+        }
+        else if (o.getLeft() instanceof GoStringLiteral && o.getRight() instanceof GoIndexOrSliceExpr) {
+          GoIndexOrSliceExpr right = (GoIndexOrSliceExpr)o.getRight();
+          GoStringLiteral left = (GoStringLiteral)o.getLeft();
+          checkTypeMismatch(o, holder, right, left, STRING_INDEX_IS_BYTE_QUICK_FIX);
+        }
+      }
+    };
+  }
+
+  private static void checkTypeMismatch(@NotNull GoConditionalExpr o,
+                                        @NotNull ProblemsHolder holder,
+                                        @NotNull GoIndexOrSliceExpr expr,
+                                        @NotNull GoStringLiteral stringLiteral,
+                                        @NotNull GoStringIndexIsByteQuickFix fix) {
+    if (isStringIndexExpression(expr)) {
+      if (isSingleCharLiteral(stringLiteral)) {
+        holder.registerProblem(o, TEXT_HINT, ProblemHighlightType.GENERIC_ERROR, fix);
+      }
+      else {
+        holder.registerProblem(o, TEXT_HINT, ProblemHighlightType.GENERIC_ERROR);
+      }
+    }
+  }
+
+  private static boolean isStringIndexExpression(@NotNull GoIndexOrSliceExpr expr) {
+    GoType type = Optional.of(expr)
+      .map(GoIndexOrSliceExpr::getExpression)
+      .map(e -> e.getGoType(null))
+      .orElse(null);
+
+    if (!GoTypeUtil.isString(type)) {
+      return false;
+    }
+
+    Trinity<GoExpression, GoExpression, GoExpression> indices = expr.getIndices();
+    PsiElement colonCandidate = Optional.ofNullable(expr.getRbrack())
+      .map(PsiElement::getPrevSibling)
+      .orElse(null);
+
+    boolean isSliceExpr = indices.getSecond() != null
+                          || indices.getThird() != null
+                          || colonCandidate.getNode().getElementType() == GoTypes.COLON;
+
+    return !isSliceExpr;
+  }
+
+  public static boolean isSingleCharLiteral(@NotNull GoStringLiteral literal) {
+    String text = literal.getText();
+    return literal.getTextLength() == 3
+           && text.charAt(0) == '\"'
+           && text.charAt(2) == '\"';
```

Comment:
Return false for "\n" or "\xff", but expected true
see https://golang.org/ref/spec#String_literals and use getDecodedText()